### ISSUE TITLE
Full Mosaic Imaging with Different Weighting Scheme

### ISSUE THREAD

**Issue by AHorneffer** [lofar-astron/factor#5]
For the different projects one wants images with weighting scheme (more natural weighting or more uniform weighting). Re-imaging the full FoV with the Factor output by hand is complicated and tedious.

My suggestion would be to extend the existing facet-re-imaging system so that either the full weighting settings, or at least the robust parameter can be specified in the Factor-parset. Also IMHO the weighting settings should become part of the pipeline name, so that re-imaging multiple times with different settings is possible without loosing the other data.

  Andreas

**Comment by darafferty** [comment]
I agree that a robust parameter for the facetimage operation would be very nice to have. One way to handle multiple runs would be to add the robust value to the operation name (e.g., facetimage_r0.2). That way, the state would not get confused (since the state is tracked using the operation name).

**Comment by mhardcastle** [comment]
At the nearby AGN meeting in Cambridge earlier this week we were
discussing a re-imaging step at significantly lower resolution to go
after low-surface-brightness stuff (eg 45 arcsec to match NVSS) which
would require tapering and cellsize changes as well as/instead of
robustness. So worth bearing those in mind as possible things that
need to be handled.

**Comment by AHorneffer** [comment]
Martin: Can you add here a list of parameters (for casapy and wsclean) that you would like to change for the final imaging?
(That way we have the parameters at hand when we get around to implement the feature.)

**Comment by darafferty** [comment]
I've started to work on this. I am thinking it's enough to allow the user to specify robust and taper values in the parset (e.g., as reimage_robust and reimage_taper_arcsec) and the cellsize is then calculated automatically from the taper value to ensure proper sampling. 

The robust and taper parameters can be lists, in which case FACTOR will loop over facetimage and mosaic operations for each set of values.

**Comment by mhardcastle** [comment]
Seems sensible to me!

**Comment by darafferty** [comment]
I've added options for the cellsize, robust, taper, and min_uv parameters (the latter for both calibration and imaging) to the reimage branch. See the example parset (examples/factor.parset) on that branch for details on how to set the options. Once the branch has been tested a bit more, I'll merge it into master.

Note that due to all these new options, I decided to reorganize the parset a bit and added [calibration] and [imaging] sections (Factor will warn you if it finds the old options in your parset but should still work).
